File: Java/CodingBat/catDog.java
// http://codingbat.com/prob/p111624

public boolean catDog(String str) {
  int numberOfCatInstances = 0;
  int numberOfDogInstances = 0;
  for (int i = 0; i < str.length()-2; ++i) {
    if (str.substring(i, i+3).equals("cat")) {
      numberOfCatInstances++;
    }
    else if (str.substring(i, i+3).equals("dog")) {
      numberOfDogInstances++;
    }
  }
  return (numberOfCatInstances == numberOfDogInstances);
}

File: Python/CodingBat/cat_dog.py
# http://codingbat.com/prob/p164876

def cat_dog(str):
  cat_count = 0
  dog_count = 0
  
  for i in range(len(str)-2):
    if str[i:i+3] == "cat":
      cat_count += 1
    elif str[i:i+3] == "dog":
      dog_count += 1
      
  return (cat_count == dog_count)
